src/__tests__/lib/api/resources/Inboxes.test.ts

```diff
@@ -0,0 +1,411 @@
+
+      const result = await inboxesAPI.delete(inboxId);
+
+      expect(mock.history.delete[0].url).toEqual(endpoint);
+      expect(result).toEqual(responseData);
+    });
+
+    it("fails with error", async () => {
+      const inboxId = 1;
+      const expectedErrorMessage = "Request failed with status code 404";
+
+      expect.assertions(2);
+
+      try {
+        await inboxesAPI.delete(inboxId);
+      } catch (error) {
+        expect(error).toBeInstanceOf(MailtrapError);
+
+        if (error instanceof MailtrapError) {
+          expect(error.message).toEqual(expectedErrorMessage);
+        }
+      }
+    });
+  });
+
+  describe("cleanInbox(): ", () => {
+    it("succesfully cleans inbox.", async () => {
+      const inboxId = 1;
+      const endpoint = `${GENERAL_ENDPOINT}/api/accounts/${accountId}/inboxes/${inboxId}/clean`;
+
+      mock.onPatch(endpoint).reply(200, responseData);
+
+      const result = await inboxesAPI.cleanInbox(inboxId);
+
+      expect(mock.history.patch[0].url).toEqual(endpoint);
+      expect(mock.history.patch[0].data).toBeUndefined();
+      expect(result).toEqual(responseData);
+    });
+
+    it("fails with error", async () => {
+      const inboxId = 1;
+      const expectedErrorMessage = "Request failed with status code 404";
+
+      expect.assertions(2);
+
+      try {
+        await inboxesAPI.cleanInbox(inboxId);
+      } catch (error) {
+        expect(error).toBeInstanceOf(MailtrapError);
+
+        if (error instanceof MailtrapError) {
+          expect(error.message).toEqual(expectedErrorMessage);
+        }
+      }
+    });
+  });
+
+  describe("markAsRead(): ", () => {
+    it("succesfully marks inbox as read.", async () => {
+      const inboxId = 1;
+      const endpoint = `${GENERAL_ENDPOINT}/api/accounts/${accountId}/inboxes/${inboxId}/all_read`;
+
+      mock.onPatch(endpoint).reply(200, responseData);
+
+      const result = await inboxesAPI.markAsRead(inboxId);
+
+      expect(mock.history.patch[0].url).toEqual(endpoint);
+      expect(mock.history.patch[0].data).toBeUndefined();
+      expect(result).toEqual(responseData);
+    });
+
+    it("fails with error", async () => {
+      const inboxId = 1;
+      const expectedErrorMessage = "Request failed with status code 404";
+
+      expect.assertions(2);
+
+      try {
+        await inboxesAPI.markAsRead(inboxId);
+      } catch (error) {
+        expect(error).toBeInstanceOf(MailtrapError);
+
+        if (error instanceof MailtrapError) {
+          expect(error.message).toEqual(expectedErrorMessage);
+        }
+      }
+    });
+  });
+
+  describe("resetCredentials(): ", () => {
+    it("succesfully resets credentials.", async () => {
+      const inboxId = 1;
+      const endpoint = `${GENERAL_ENDPOINT}/api/accounts/${accountId}/inboxes/${inboxId}/all_read`;
+
+      mock.onPatch(endpoint).reply(200, responseData);
+
+      const result = await inboxesAPI.markAsRead(inboxId);
```

Comment:
This should test `resetCredentials` but tests `markAsRead`

```suggestion
  describe("resetCredentials(): ", () => {
    it("succesfully resets credentials.", async () => {
      const inboxId = 1;
      const endpoint = `${GENERAL_ENDPOINT}/api/accounts/${accountId}/inboxes/${inboxId}/all_read`;

      mock.onPatch(endpoint).reply(200, responseData);

      const result = await inboxesAPI.markAsRead(inboxId);
```